### PR TITLE
bpo-45873: Restore Python 3.6 compatibility (GH-29730)

### PR DESCRIPTION
**Pull request by tiran**
https://bugs.python.org/issue45873

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]
LGTM, but is it even worth testing for 3.7+? Why not always take the slow path?

**Comment by tiran** [comment]
> LGTM, but is it even worth testing for 3.7+? Why not always take the slow path?

I left the old implementation in because it's more readable. The fast path is now in a doc string for future reference.

I also added more debug code to configure to figure out what is going on on the Debian buildbot.

**Comment by tiran** [comment]
Closing and reopening to retrigger Azure.

**Review by gvanrossum — APPROVED** [review]
(Why do we even bother with "python3 python"? For weird platforms that have only one of those but it happens to be 3.6 or better?)

**Comment by tiran** [comment]
> (Why do we even bother with "python3 python"? For weird platforms that have only one of those but it happens to be 3.6 or better?)

I don't know and kept ``python 3 python`` just in case.

File: Tools/scripts/deepfreeze.py [hunk 1]
```diff
@@ -5,18 +5,38 @@
...
+    return b.isascii() and b.decode("ascii").isprintable()
+    """
+    try:
+        s = b.decode("ascii")
+    except UnicodeDecodeError:
+        return False
+    for c in s:
+        if unicodedata.category(c) in NON_PRINTABLE:
+            return False
+    return True
```
Comment (gvanrossum) [inline Tools/scripts/deepfreeze.py]: What I really wanted is this:

```suggestion

def isprintable(b: bytes) -> bool:
    return all(0x20 <= c < 0x7f for c in b)
```

There's only one call site and it just uses this to decide whether to use hex escapes for string literals in the generated output. Unicode character classes feel like overkill. :-)

**Comment by tiran** [comment]
Broke Azure tests are likely caused by GH-28612.

File: Tools/scripts/deepfreeze.py (hunk 1 above)
Comment (tiran) [reply Tools/scripts/deepfreeze.py]: $deity bless ASCII-only land! It's so much easier. :)

**Review by gvanrossum — APPROVED** [review]
:-)

**Comment by gvanrossum** [comment]
> Broke Azure tests are likely caused by [GH-28612](https://github.com/python/cpython/pull/28612).

How? That PR was closed without merging. And if there was a cause outside this PR why would it not have failed before (e.g. when I submitted #29717)?

Are you suggesting there are some cached files from running that PR present?

**Comment by tiran** [comment]
I confused GH-28612 with GH-29488. Both are for bpo-39026.

This PR was created after @vstinner merged GH-29488.

**Pull request merged**